core/src/main/java/dk/alexandra/fresco/framework/builder/binary/BinaryBuilder.java

```diff
@@ -0,0 +1,88 @@
+package dk.alexandra.fresco.framework.builder.binary;
+
+import dk.alexandra.fresco.framework.Computation;
+import dk.alexandra.fresco.framework.value.SBool;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Interface for the basic operations which any binary protocol suite needs to implement.
+ * 
+ * @author Kasper Damgaard
+ *
+ */
+public interface BinaryBuilder {
+
+  Computation<SBool> known(boolean known);
+
+  default List<Computation<SBool>> known(boolean[] known) {
```

Comment:
Known should be used for fixed constants in the calculations. Returning a list is weird, since boolean arithmetic only has two constants. 

In this case it is just testing that is made easy. I would suggest either to use input in those test, move this implementation to test or even stream the arrays (comp1.stream().map(builder::known).collect(...) )